File: 코드/ch8/clip2/boj9935/main.py
import sys

def main():
    str_ = sys.stdin.readline().rstrip()
    bomb = sys.stdin.readline().rstrip()

    stack = []
    for ch in str_:
        stack.append(ch)
        if len(stack) < len(bomb):
            continue

        is_same = True
        for i in range(len(bomb)):
            if stack[-len(bomb) + i] != bomb[i]:
                is_same = False
                break
        if is_same:
            for _ in range(len(bomb)):
                stack.pop()

    result = ''.join(stack)
    if not result:
        sys.stdout.write("FRULA\n")
    else:
        sys.stdout.write(result + '\n')

if __name__ == "__main__":
    main()

File: 코드/ch8/clip2/boj9935/Main.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.next();
        String bomb = sc.next();

        int n = str.length();
        int m = bomb.length();

        Stack<Character> s = new Stack<>();
        for(int i=0; i<n; i++) {
            s.add(str.charAt(i));
            if(s.size() < m) continue;

            boolean isSame = true;
            for(int j=0; j<m; j++) {
                if(s.get(s.size() - m + j) != bomb.charAt(j)) {
                    isSame = false;
                    break;
                }
            }
            if(isSame) {
                for(int j=0; j<m; j++) s.pop();
            }
        }
        StringBuilder sb = new StringBuilder();
        for(char ch : s) {
            sb.append(ch);
        }
        if(sb.length() == 0) System.out.println("FRULA");
        else System.out.println(sb);
    }
}
